routing.asciidoc

```diff
@@ -16,112 +16,180 @@ Finally, you will learn about the SPHINX mix Format that enables onion routing w
+We could amend the contract so that if it is not fulfilled by a certain deadline, then the contract expires and the escrow service returns the money to the person who made the original deposit.
+We call this deadline a "time lock" as the deposit is locked with the escrow service for a certain amount of time, and is eventually released even if no proof of payment was provided.
+Alice could add a line:
 
-_Bob has 24 hours to show the secret after the contract was signed. If the time has passed Alice will get her deposit back from the escrow service and the contract becomes invalid._
+[alice-gloria-routing-7]
+----
+_Bob has 24 hours to show the secret after the contract was signed.
+If he does not provide the secret by this time, Alice's deposit will be refunded by the escrow service and the contract becomes invalid._
+----
 
-Bob of course now has to make sure to get receive the proof of payment faster than in 24 hours.
-If he would receive a later proof the contract between him and Alice would already be nullified and Bob would not get reimbursed.
-After he signed the contract with Alice he alters the original contract between him and Wei in the following way:
+Bob, of course, now has to make sure he receives the proof of payment within 24 hours.
+Even if he successfully pays Wei, if he receives the proof of payment later than 24 hours he will not be reimbursed.
+In turn, he will alter his contract with Wei in the following way:
 
-_Wei has 22 hours to show the secrete after the contract was signed. If the time has passed Bob will get his deposit back from the escrow service and the contract becomes invalid._
+[alice-gloria-routing-7]
+----
+_Wei has 22 hours to show the secret after the contract was signed.
+If he does not provide the secret by this time, Bob's deposit will be refunded by the escrow service and the contract becomes invalid._
+----
 
-As you have guessed Wei is now incentiviced to also alter his contract with Gloria:
+As you might have guessed, Wei is now incentiviced to also alter his contract with Gloria:
 
-_Gloria has 20 hours to show the secrete after the contract was signed. If the time has passed Wei will get his deposit back from the escrow service and the contract becomes invalid._
+[alice-gloria-routing-9]
+----
+_Wei has 20 hours to show the secret after the contract was signed.
+If he does not provide the secret by this time, Bob's deposit will be refunded by the escrow service and the contract becomes invalid._
+----
 
-With such a chain of contracts we can be sure that after 24 hours of setting up the first contract that the payment was either successfully delivered from Alice via Bob and Wei to Gloria or that the payment has failed and was not conducted at all.
+With such a chain of contracts we can ensure that, after 24 hours, the payment will succesfully deliver from Alice to Bob to Wei to Gloria, or it will fail and everyone will be refunded.
 It cannot be stuck in the middle of the road.
```

Comment:
This part I am not sure about. Consider the corner case:

Gloria reveals the secret to Wei, gets 10 gold coins
Wei reveals the secret to Bob, gets 11 gold coins
Bob's node crashes, goes offline, and he cannot reveal the secret to Alice by the time the HTLC expires.
The HTLC expires, the escrow ends, Alice is refunded her 12 gold coins.

Gloria ends up with 10, Wei ends up with 1, and Alice still has her 12 coins.
Bob paid Wei 11, but did not receive 12 from Alice so he's lost 11 coins and essentially picked up the tab.
What now?